contracts/contracts/EthereumUtils.sol

```diff
@@ -203,24 +222,63 @@ library EthereumUtils {
         internal
         view
         returns (
-            address pubkey_addr,
-            bytes32 r,
-            bytes32 s,
-            uint8 v
+            address pubkeyAddr,
+            bytes32 sigR,
+            bytes32 sigS,
+            uint8 sigV
```

Comment:
This looks like a `SignatureRSV`. Was there a reason not to use it? If it's to avoid encoding the struct, I'd first check whether solc optimizes that out (e.g. using the `viaIR` flag) before writing hand-optimized code.